### PR TITLE
added minibatch gradient descent

### PR DESCRIPTION
**Pull request by channiemills**

### CODE REVIEW

**Comment by channiemills** [comment]
Added minibatch gradient descent in with existing algorithms. Also included a keras nn with parameters matching hand coded full batch gradient descent but time to converge seemed long.
